src/main/scala/mesosphere/marathon/core/task/termination/impl/KillServiceActor.scala

```diff
@@ -88,6 +91,23 @@ private[impl] class KillServiceActor(
     case KillInstancesAndForget(instances) =>
       killInstances(instances, None)
 
+    case goalChange: InstanceGoalChanged if shouldBeKilled(goalChange.goal) =>
+      if (instancesToKill.contains(goalChange.id)) {
+        logger.info(s"Ignoring goal change to ${goalChange.goal} for ${goalChange.id} since the instance is already queued.")
+      } else {
+        logger.info(s"Adding ${goalChange.id} to the queue since its goal changed to ${goalChange.goal}")
+        killInstances(Seq(goalChange.instance), maybePromise = None)
+      }
+
+    case goalChange: InstanceGoalChanged =>
+      // if the goal changed to something != Stopped/Decommissioned, we might
+      // want to take the instance out of the inFlight
+      // note: a goal change from Decommissioned should not be possible
+      if (instancesToKill.contains(goalChange.id) || inFlight.contains(goalChange.id)) {
+        // TODO: stop trying to kill this instance
```

Comment:
Is this addressed in this PR? Why are our tests passing without that change?

Reply:
I don't think this can actually happen right now. if we set goal back to running it's after the tasks were killed so we don't have to stop killing.

I would assume that logging this is fine for now...

Reply:
TODO: remove comment
keep log
make sure this operation is illegal to move out of decommissioned
for stopped it is OK, make sure that's understood